src/TableFunctions/TableFunctionExplain.h

```diff
@@ -24,7 +24,7 @@ class TableFunctionExplain : public ITableFunction
 
     InterpreterExplainQuery getInterpreter(ContextPtr context) const;
 
-    ASTPtr query = nullptr;
+    std::shared_ptr<ASTExplainQuery> query = nullptr;
```

Comment:
In `parseArguments` function you can create shard ptr of `ASTExplainQuery` and set it to ASTPtr only at the end of function to avoid changing this type from `ASTPtr`.